warehouse/templates/pages/security-key-giveaway.html

```diff
@@ -0,0 +1,205 @@
+      </p>
+      <p>
+        Eligible maintainers will be able to redeem a promo code for two
+        free <a
+        href="https://store.google.com/product/titan_security_key">Titan
+        Security Keys</a> (either USB-C or USB-A), including free shipping.
+      </p>
+
+      <div class="callout-block">
+      <h2>Determine your eligibility</h2>
+      {% if promo_code %}
+        {# Show the user their promo code #}
+        <p>
+        Success! Your promo code is:
+        </p>
+        <p class="copyable">
+            <span id="promo-code">{{ promo_code.code }}</span>
+            <button type="button" class="copy-tooltip copy-tooltip-s" data-clipboard-target="#promo-code" data-tooltip-label="Copy to clipboard">
+              <i class="fa fa-copy" aria-hidden="true"></i>
+              <span class="sr-only">Copy promo code</span>
+            </button>
+        </p>
+        <p>
+        You can order your keys via <a href="https://store.google.com/product/titan_security_key">the Google Store</a> by applying this promo code at checkout. This code expires on May 1, 2022.
+        </p>
+      {% elif not request.user %}
+        {# The user is logged out, ask them to log in #}
+        <a href="{{ request.route_url("accounts.login", _query={REDIRECT_FIELD_NAME: request.path_qs}) }}" class="button button--primary">
+          Log in to determine if you are eligible
+        </a>
+      {% elif not eligible %}
+        {# The user is not eligible, tell them why #}
+        <p>Sorry, you're not currently eligible for a promo code.</p>
+        <p><b>Reason:</b> {{ reason_ineligible }}</p>
+      {% else %}
+        {# The user is eligible, show the form #}
+        <p>
+          Congrats, you may be eligible to receive a promo code! The following projects you maintain are considered critical projects:
+        </p>
+        <ul>
+          {% for project in eligible_projects %}
+          <li>{{ project }}</li>
+          {% endfor %}
+        </ul>
+        <p>
+          We just need a little more information. (<a href="#regions">Why?</a>)
+          <form method="POST">
+            <input name="csrf_token" type="hidden" value="{{ request.session.get_csrf_token() }}">
+            {{ form.country }}
+            <button type="submit" class="button button--primary">Request a promo code</button>
+          </form>
+        </p>
+      {% endif %}
+      </div>
+
+
+      <section class="faq-group">
+      <h2 id="faq">FAQ</h2>
+      <p>Answers to frequently asked questions regarding this effort:</p>
+
+      <h3 id="critical-eligiblity">What determines if project is a critical project?</h3>
+      <p>
+        PyPI determines project eligibility based on download counts derived
+        from PyPI's <a
+        href="https://warehouse.pypa.io/api-reference/bigquery-datasets.html">public
+      dataset of download statistics</a>. Any project in the top 1% of
+    downloads over the prior 6 months is designated as critical.
```

Comment:
are we not planning to include warehouse dependencies?

Reply:
I considered it but thought that this might open up a broader question of whose dependencies are important, e.g. are warehouse dependencies more critical than twine dependencies? wheel? etc.

Reply:
Based on the centralized nature of PyPI, I would argue that yes it is more important to protect in depth. A compromise of PyPI that is undetected for any length of time could have catastrophic impacts downstream. Installers are probably next highest priority, and then upload tools by the metric of "how bad how fast".

Reply:
Fine with me, should be trivial to add this to the task, as long as we're OK taking a runtime dependency on https://github.com/di/pip-api 😉